src/Coupon/CouponSyncer.php

```diff
@@ -234,30 +240,36 @@ public function delete( DeleteCouponEntry $coupon ) {
 		$invalid_promotions = [];
 		$synced_google_ids  = $coupon->get_synced_google_ids();
 		$wc_coupon          = $this->wc->maybe_get_coupon(
-			$coupon->get_coupon()
-				->get_wc_coupon_id()
+			$coupon->get_wc_coupon_id()
 		);
 		$wc_coupon_exist    = $wc_coupon instanceof WC_Coupon;
 		foreach ( $synced_google_ids as $target_country => $google_id ) {
 			try {
-				$adapted_coupon = $coupon->get_coupon();
+				$adapted_coupon = $coupon->get_google_promotion();
 				$adapted_coupon->setTargetCountry( $target_country );
-				$adapted_coupon->disable_coupon();
 
+				do_action(
+					'woocommerce_gla_debug_message',
+					sprintf(
+						'Start to delete coupon (ID: %s) as promotion structure: %s',
+						$coupon->get_wc_coupon_id(),
+						json_encode( $adapted_coupon )
+					),
+					__METHOD__
+				);
 				$response = $this->google_service->create( $adapted_coupon );
```

Comment:
This seems a bit confusing. Could you please add a comment here indicating that there's no method to `delete` a coupon; therefore, we need to update the existing one with an expiry date.

Reply:
updated in comments in the latest version of the code